### DISCUSSION TITLE
Issues with AioRTC Async and Flask

### DISCUSSION THREAD

**Discussion by R0NAM1** [aiortc/aiortc#1088] [Q&A]
Hi all,

I'm attempting to create a mass use Camera To WebRTC (Or RTSP to WebRTC) program in python and I can get the initial connection going just fine, but stopping it after a client disconnects is difficult, I have a webrtc datachannel that if a message isn't received from the browser in 5 seconds it runs the following closing function:

(https://pastebin.com/Ku9NkgNF)

The following is the way I create my WebRTC sessions with AioRTC:

(https://pastebin.com/rSPBFygj)

And the issue I am having is that the AioRTC RTP objects do not go away on their own, and any attempt to call RTCPeerConnection.close() always hangs forever, so it slowly takes up CPU threads and memory. I know this because when I force quit the program long after they should be dead I get the following:

(https://pastebin.com/3VpSa4XZ)

**Comment by ZoroLH** [comment]
Consider this #1089 as well.

Seams like the pyav close is not working.  In MediaPlayer, av.open() to start ffmpeg.  I managed to call _stop method and run self.__container.close() self.__container = None.    Memeory is till not released.   Seams like pyav's issue.

**Reply by ZoroLH** [reply]
With MediaRelay().subscribe(webcam.video, buffered=False), memory usage is 200 M on first connection, and extra couple M on other connection, which is OK. But from second connection, FPS is realy low, seams like less than 10 FPS.

**Reply by ZoroLH** [reply]
How to close pyav: https://github.com/PyAV-Org/PyAV/issues/370

**Comment by ZoroLH** [comment]
Related pyav issues are here:  https://github.com/PyAV-Org/PyAV/issues/1117
https://github.com/PyAV-Org/PyAV/issues/751
Which seams to be fixed already, can some body upgrade the pyav's version in this project?
